File: server/models/track.go
package models

type Track struct {
    Id       string   `bson:"_id,omitempty" json:"id"`
    Name     string   `bson:"name" json:"name"`
    Artist   string   `bson:"artist" json:"artist"`
    Text     string   `bson:"text" json:"text"`
    Listens  int      `bson:"listens" json:"listens"`
    Picture  string   `bson:"picture" json:"picture"`
    Audio    string   `bson:"audio" json:"audio"`
    Comments []string `bson:"comments" json:"comments"`
}

type TrackPaginationParams struct {
    Count   int
    Offset  int
}

File: client/src/types/track.ts
import { StaticImageData } from "next/image";

export interface TrackProps{
    track: TrackResp;
    isFavourite: boolean;
    // duration: number; // will be past from back
}

export interface TrackResp {
    artist: string;
    audio: string;
    comments: any; //will be deleted
    id: string;
    listens: number;
    name: string;
    picture: string;
    text: string;
    //duration: number;
}

export interface TracksResp {
    tracksData: TrackResp[];
}

export interface TrackState {
    tracks: TrackResp[]
    error: string
}

export enum TrackActionTypes {
    FETCH_TRACKS = 'FETCH_TRACKS',
    FETCH_TRACKS_ERROR = 'FETCH_TRACKS_ERROR',
}

interface FetchTracksAction {
    type: TrackActionTypes.FETCH_TRACKS
    payload: TrackResp[]
}

interface FetchTracksErrorAction {
    type: TrackActionTypes.FETCH_TRACKS_ERROR
    payload: string
}

export type TrackAction = 
    FetchTracksAction
    |   FetchTracksErrorAction